citesphere/src/main/java/edu/asu/diging/citesphere/core/service/jobs/impl/SyncJobManager.java

```diff
@@ -74,13 +75,24 @@ public long getJobsCount(IUser user) {
     }
     
     @Override
-    public void cancelJob(String jobId) {
+    public void cancelJob(String jobId) throws InterruptedException {
         Optional<GroupSyncJob> jobOptional = jobRepo.findById(jobId);
         if (jobOptional.isPresent()) {
             GroupSyncJob job = currentJobs.get(jobOptional.get().getGroupId());
             if (job == null) {
                 job = jobOptional.get();
             }
+            long threadId = job.getThreadId();
+            
+            Set<Thread> setOfThreads = Thread.getAllStackTraces().keySet();
+
+            //Iterate over set to find the current thread id
+            for(Thread thread : setOfThreads){
+                if(thread.getId() == threadId){
+                    thread.interrupt();
+                }
+            }
+            
```

Comment:
This seems a little too hacky and I'm wondering what issues we might run into. Also, we want to make sure, we don't get stuck with partial data, e.g. the job should accurately reflect how many items have been processed. I would try the approach described here in the answer: https://stackoverflow.com/questions/38880069/spring-cancel-async-task

- return a asnycResult object in the async method
- store the results object somewhere in memory
- when canceling a task, call `cancel` on the matching result object
- check `Thread.currentThread().isInterrupted()` throughout the async method and cleanly shutdown the sync if interrupt has been called.